bundle/src/main/java/com/adobe/acs/tools/dumplibs/DumplibsServlet.java

```diff
@@ -0,0 +1,217 @@
+package com.adobe.acs.tools.dumplibs;
+
+import com.adobe.granite.ui.clientlibs.ClientLibrary;
+import com.adobe.granite.ui.clientlibs.HtmlLibrary;
+import com.adobe.granite.ui.clientlibs.HtmlLibraryManager;
+import org.apache.felix.scr.annotations.Reference;
+import org.apache.felix.scr.annotations.sling.SlingServlet;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.servlets.SlingSafeMethodsServlet;
+import org.apache.sling.commons.json.JSONArray;
+import org.apache.sling.commons.json.JSONException;
+import org.apache.sling.commons.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+@SlingServlet(
+        label = "ACS AEM Tools - Dumplibs",
+        description = "ACS AEM Tools End-point for Dumplibs",
+        methods = {"GET"},
+        paths = {"/bin/acs-tools/dumplibs"},
+        extensions = {"json"},
+        metatype = true
+)
+public class DumplibsServlet extends SlingSafeMethodsServlet {
+
+    private static final Logger log = LoggerFactory.getLogger(DumplibsServlet.class);
+    private static final String CONTENT_TYPE_JSON = "application/json";
+
+    @Reference
+    HtmlLibraryManager libraryManager;
+
+    @Override
+    protected final void doGet(SlingHttpServletRequest request, SlingHttpServletResponse response)
+            throws ServletException, IOException {
+
+        DumplibsParams p = new DumplibsParams(request);
+
+        // if path is present in request params, it takes precedence
+        // get clientlib by path
+        if (p.getPath() != null) {
+            handlePathRequest(request, response, p);
+        }
+        // If path is NOT present and categories is present in request params
+        // get clientlibs by categories
+        else if (p.getCategories() != null) {
+            handleCategoriesRequest(request, response, p);
+        }
+        // Neither path nor categories are present in request params
+        // get all clientlibs
+        else {
+            handleAllLibsRequest(request, response, p);
+        }
+
+    }
+
+    /**
+     * Handles request to get clientlib info by path
+     */
+    private void handlePathRequest(SlingHttpServletRequest request, SlingHttpServletResponse response, DumplibsParams p)
+            throws IOException {
+        HtmlLibrary lib = libraryManager.getLibrary(p.getType(), p.getPath());
+        try {
+
+            JSONObject libJSON = htmlLibraryToJSON(lib); // should never be null
```

Comment:
Sling JSON is deprecated :( .. were moving over to GSON (which is provided by AEM) or Sling Model Exporter + Jackson .. i assume this will be easier to switch to GSON than SME.

Reply:
@davidjgonzalez I dont know that GSON is available in older AEM versions and I do know that Sling JSON is deprecated, I chose it for backwards compatibility and because other code in this repo uses it. 

So how do you think we should handle including GSON dependency? Should this project include GSON? should it be provided? if so which version of GSON is included in which versions of AEM?

Reply:
ACS Tools now only supported on AEM 6.2+ -- which i'm fairly certain includes the GSON dependency (since that's what we've moved to for ACS Commons which also supports 6.2+)

You can use the same as what's used in ACS Commons and we should be good (and future proof).
https://github.com/Adobe-Consulting-Services/acs-aem-commons/blob/master/bundle/pom.xml#L247

Reply:
@davidjgonzalez moved to GSON as per your recommendation, see my latest commits :)